tests: Refactor test configs

mkjost0 opened the pull request:
The yaml file changes made here will need to be copied to develop for our scheduled tests to run properly.  In addition, the original comments on this can be seen here:  https://gem5-review.googlesource.com/c/public/gem5/+/70340/2

mkjost0 committed:
tests: Add READMEs to the testing directory

This change adds READMEs to each directory within tests/gem5,
with a short description of the test, as well as how to run it.

Change-Id: I574ebcdc837848b52f21e8c0f8856ff09463284b

mkjost0 committed:
tests: Update layout of testing directory

This changeset reorganizes the testing directory within gem5,
removing the bigger config folders, then replacing them with
smaller configs folders within each directory containing only
the scripts necessary for that set of tests. It also changes
the locations of the config scripts used in each set of tests,
and updates the tests accordingly.

Change-Id: I38297d4496f72bd5cf7200471acd5c4d93002b27

mkjost0 committed:
tests: Unify testing directory names

This updates the testing directory so everything uses underscores
and is more uniform.

Change-Id: I7ea45c9e0fc1892605387cd2453ce8656ddccd49

mkjost0 committed:
misc: Update test names in daily and weekly yaml files

Updates the directories in which tests are run in accordance
with the refactoring of the testing directory

Change-Id: I93f5c5b0236c5180da04deb425ec2ed6804fa003

powerjg commented on tests/gem5/asmtest/README.md:
````diff
@@ -0,0 +1,9 @@
+To run these tests by themselves, you can run the following command in the tests directory:
+
+```bash
+./main.py run gem5/asmtest --length=[length]
````
What does `--length` mean in this case?

I guess this is copy pasted across all of these readme's, but does it make sense for this one?

I suggest listing the tests in these readme and saying whether they are quick/long/very long.   HOWEVER, I suggest doing that in another change 😄

powerjg commented on tests/gem5/kvm_switch_tests/README.md:
```diff
@@ -0,0 +1,8 @@
+
+These tests ensure that gem5 can switch processors during simulation.
+To run these tests by themselves, you can run the following command in the tests directory:
+
```
Similar to above, in another PR, it would be good to specify the system requirements to run these tests.

mkjost0 replied:
Sounds good, I can make the extra changes to the READMEs after this PR gets in

powerjg commented on tests/gem5/stdlib/configs/simple_binary_run.py:
How much of this code is duplicated in other files? 

There's a tradeoff between reducing code duplication and complexity. But I worry a bit about how much we are going to have to refactor in the tests if something changes.

mkjost0 replied:
Some of the code is duplicated, as every instance of simple_binary_run.py in the testing directory started from the same one file.  However, splitting it up got rid of a lot of the optional parameters that used to be at the top, and duplicating it makes it easier to see exactly what's being run in each directory.

powerjg reviewed (commented):
Just some comments/questions and notes for the future. This looks good to me. I'll let Bobby approve, though.

mkjost0 committed:
tests: Add checkpoint.py to fs tests directory

The configs directory for the fs tests was missing the
checkpoint.py file, causing some of the CI tests to fail.

Change-Id: Ifbd775ad658f96d06bea7bee554fe3bedcf5a5b5

mkjost0 committed:
Merge branch 'develop' into refactor-test-configs

mkjost0 committed:
tests: Add switcheroo.py to fs configs directory

There was another missing file in the configs for the fs tests,
so this should allow the switcheroo tests to pass.

Change-Id: Ic4e26cceeb9209f176158b80eaaba88b47968c39
